go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -550,6 +551,14 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 			buf := make([]byte, 1<<16) // 64 KB buffer size
 			stackSize := runtime.Stack(buf, true)
 			log.Errorf("Stack trace:\n%s", string(buf[:stackSize]))
+			// This select is only to handle the race, where we start to set the demote primary stalled
+			// but then the function finishes. So, after we set demote primary stalled, we check if the
+			// function has finished and if it has, we clear the demote primary stalled.
+			select {
+			case <-finishCtx.Done():
```

Comment:
Wouldn't a check against `finishCtx.Err()` be enough here (instead of a `select`)? 🤔

Reply:
Yeah, I don't know what I was thinking 😆